src/main/js/controls/Graph/helpers/helpers.js

```diff
@@ -865,6 +865,7 @@ const removeNoDataView = (svg) => {
  * @returns {d3.selection} d3 selection node of svg.
  */
 const drawNoDataView = (config, svg) => {
+    removeNoDataView(svg);
```

Comment:
We need to remove the existing no data view before we draw the no data view. Otherwise, it will draw the no data view above the existing one, and as a result tick values and gridlines gets dithered on each panning click if legend holds no data in every case.